tools/istio-docker.mk

```diff
@@ -38,18 +38,19 @@ GRAFANA_FILES:=addons/grafana/start.sh \
                addons/grafana/import_dashboard.sh
 
 # note that "js" and "force" are directories rather than a file
-$(ISTIO_DOCKER)/js: addons/servicegraph/js | $(ISTIO_DOCKER)
-	cp -r $< $(@D)
-
-$(ISTIO_DOCKER)/force: addons/servicegraph/force | $(ISTIO_DOCKER)
+$(ISTIO_DOCKER)/js $(ISTIO_DOCKER)/force: addons/servicegraph/$$(notdir $$@) | $(ISTIO_DOCKER)
 	cp -r $< $(@D)
 
 # generated content
```

Comment:
A bit unrelated to this PR - while I like the more consistent Dockerfile naming, the current scheme is pretty inefficient,
the common directory where all binaries are copied results in a lot of copy, docker build needs to copy all the 
files from that dir even if each package only needs 2-3. 

I would rather have some duplication but clear targets instead of string replacements and macros..
This would also allow us to further optimize the pipeline, by building only the subset of docker images needed
by the tests, or doing builds in parallel.

Reply:
I had pondering sticking each docker image in its own directory (e.g., a subdirectory of ISTIO_DOCKER based on the image name), but was taking incremental steps towards that.  It might force a few of the copy rules to have some duplication, though I've also considered a scheme where each image just has a list of dependencies and something like filter/filter-out is used to pick out the Dockerfile (match on "Dockerfile"), stuff to be built (match on "$(ISTIO_OUT)"), and stuff to be copied from source (anything that doesn't match on "$(ISTIO_OUT)").  It'll take some experimentation to figure out what does or doesn't work well, but first achieving consistency seemed like a way to make optimization more straightforward.